.circleci/config.yml

```diff
@@ -44,17 +44,33 @@ jobs:
             for demo in !($IGNORED_DIRS); do
               if [ -d "$demo" ]; then
                 (
+
+                  # Call pyenv if we are using MacOS
+                  os="$(uname -s)"
+                  if [ "$os" == "Darwin" ]; then
+                    eval "$(pyenv init -)"
+                    pyenv local $PYTHON
+                  fi
+
+                  # Create demo specific virtual environments
                   env="envs/$demo"
                   python -m virtualenv "$env"
                   . "$env/bin/activate"
 
                   pip install -r "$demo/requirements.txt"
                   pip install coverage
+
+                  # Python 2.7 case
+                  # Note: using 2>&1 because python 2.7 sends version info to stderr
+                  python_version="$(python --version 2>&1)"
+                  if [[ "$python_version" =~ ^Python\ 2\.7\.[0-9]+$ ]]; then
```

Comment:
Bash pro tip: you can have regex in a variable to avoid escaping:
```
regex='^Python 2.7.[0-9]+$'
[[ $python_version =~ $regex ]] && pip install mock
```

(Also, don't need to quote vars inside `[[ ]]`. And here a simpler expression would be enough: `[[ $pyver =~ "Python 2.7" ]]`.)

Reply:
But, looking at a bigger picture, you would want to have this requirement (for Python 2.7) in the requirements file, to simplify the build script, e.g:
```
mock; python_version=="2.7"
```
Check [an example](https://github.com/dwavesystems/dwave-cloud-client/blob/master/requirements.txt).